compose/env-file.md

```diff
@@ -5,8 +5,10 @@ title: Declare default environment variables in file
 ---
 
 Compose supports declaring default environment variables in an environment file
-named `.env` placed in the project directory. The project directory is specified by order of precedence:
-- `--project-dir` flag
+named `.env` placed in the project directory. Versions of Docker Compose prior to `1.28`, load the `.env` from the current working directory where the command is executed, or, the project directory if this is explicitly set with the `--project-directory` argument. This inconsistency has been addressed since `+v1.28` by limiting the `.env` file path to the project directory.
+
```

Comment:
you could add here a note that users relying on this inconsistent behaviour have a simple workaround by using --env-file explicitly